model/bi/webhook.go

```diff
@@ -17,52 +21,158 @@ import (
+	case EventAccountEnabled, EventAccountDisabled:
+		return c.handleAccountEnabledOrDisabled()
+	}
+	return errors.New("event not handled")
+}
+
+func (c *WebhookCall) checkToken() error {
+	for _, acc := range c.accounts {
+		if acc.ID() == aggregatorID {
+			if subtle.ConstantTimeCompare([]byte(c.Token), []byte(acc.Token)) == 1 {
+				return nil
+			}
+			return errors.New("token is invalid")
+		}
+	}
+	return errors.New("no bi-aggregator account found")
+}
+
+func (c *WebhookCall) handleConnectionSynced() error {
+	connID, err := extractPayloadConnID(c.Payload)
 	if err != nil {
 		return err
 	}
 	if connID == 0 {
 		return errors.New("no connection.id")
 	}
 
-	account, err := findAccount(accounts, connID)
+	uuid, err := extractPayloadConnectionConnectorUUID(c.Payload)
+	if err != nil {
+		return err
+	}
+	slug, err := mapUUIDToSlug(uuid)
 	if err != nil {
+		c.Instance.Logger().WithNamespace("webhook").
+			Warnf("no slug found for uuid %s: %s", uuid, err)
 		return err
 	}
-	trigger, err := findTrigger(inst, account)
+	konn, err := app.GetKonnectorBySlug(c.Instance, slug)
+	if err != nil {
+		userID, _ := extractPayloadUserID(c.Payload)
+		c.Instance.Logger().WithNamespace("webhook").
+			Warnf("konnector not installed id_connection=%d id_user=%d uuid=%s slug=%s", connID, userID, uuid, slug)
+		return nil
+	}
+
+	var trigger job.Trigger
+	account, err := findAccount(c.accounts, connID)
+	if err != nil {
+		account, trigger, err = c.createAccountAndTrigger(konn, connID)
+	} else {
+		trigger, err = findTrigger(c.Instance, account)
+	}
 	if err != nil {
 		return err
 	}
 
-	if mustExecuteKonnector(inst, trigger, payload) {
-		return fireTrigger(inst, trigger, account, payload)
+	if c.mustExecuteKonnector(trigger) {
+		return c.fireTrigger(trigger, account)
 	}
-	return copyLastUpdate(inst, account, payload)
+	return c.copyLastUpdate(account)
 }
 
-func checkToken(accounts []*account.Account, token string) error {
-	for _, acc := range accounts {
-		if acc.ID() == aggregatorID {
-			if subtle.ConstantTimeCompare([]byte(token), []byte(acc.Token)) == 1 {
-				return nil
-			}
-			return errors.New("token is invalid")
-		}
+func mapUUIDToSlug(uuid string) (string, error) {
+	f := statik.GetAsset("/mappings/bi-banks.json")
+	if f == nil {
+		return "", os.ErrNotExist
 	}
-	return errors.New("no bi-aggregator account found")
+	var mapping map[string]string
+	if err := json.Unmarshal(f.GetData(), &mapping); err != nil {
+		return "", err
+	}
+	slug, ok := mapping[uuid]
+	if !ok || slug == "" {
+		return "", errors.New("not found")
+	}
+	return slug, nil
+}
+
+func extractPayloadConnectionConnectorUUID(payload map[string]interface{}) (string, error) {
+	conn, ok := payload["connection"].(map[string]interface{})
+	if !ok {
+		return "", errors.New("connection not found")
+	}
+	connector, ok := conn["connector"].(map[string]interface{})
+	if !ok {
+		return "", errors.New("connection.connector not found")
+	}
+	uuid, ok := connector["uuid"].(string)
```

Comment:
the connector's uuid can be found in connection.connector_uuid and not in connection.connector.uuid